indispensable/src/commonMain/kotlin/at/asitplus/signum/indispensable/pki/X509Certificate.kt

```diff
@@ -63,7 +92,20 @@ constructor(
     val issuerAlternativeNames: AlternativeNames? = extensions?.findIssuerAltNames()
 
 
-    private fun Asn1TreeBuilder.Version(value: Int) = Asn1.ExplicitlyTagged(Tags.VERSION.tagValue) { +Asn1.Int(value) }
+    private fun Asn1TreeBuilder.Version(value: Int) =
+        Asn1.ExplicitlyTagged(Tags.VERSION.tagValue) { +Asn1.Int(value) }
+
+    val keyUsage: Set<X509KeyUsage>
+        get() = extensions
+            ?.find { it.oid == KnownOIDs.keyUsage }
+            ?.value
+            ?.asEncapsulatingOctetString()
+            ?.children
+            ?.getOrNull(0)
+            ?.let { it as? Asn1Primitive }
+            ?.let(Asn1BitString::decodeFromTlv)
+            ?.let(X509KeyUsage::doDecode)
+            ?: emptySet()
```

Comment:
are we sure this isn't to permissive by just falling back to the empty set (instead of throwing) on any unexpected behavior? For example, is it valid for the thing to be present but not be an Asn1Primitive?